Improvement/2149 documentation access from the ui

ChengYanJin opened the pull request:
**Component**: ui, doc

**Context**: 
We would like to access the doc from UI.

**Summary**:
- Add the documentation entry point from the Navbar 
- Add the link in the volume creation page
- Change the icon in clusterMonistering page ( question icon => infomation icon)
- Fix bugs for create volume list 

**Acceptance criteria**: 
Add the link in the volume creation page and add the documentation entry point from the Navbar:

![image](https://user-images.githubusercontent.com/18453133/71962977-f5952a00-31fa-11ea-95c7-8a83c72de495.png)

- should pass the cypress test:
`./node_modules/cypress/bin/cypress run -b chrome -s cypress/integration/AccessDocumentation.feature`

---

<!-- Declare one or more issues to close once this PR gets merged -->

Closes: #2149 

The integration branch in 2.5 is:
 `w/2.5/improvement/2149-documentation-access-from-the-UI`

ChengYanJin committed:
ui/doc: access docs from UI

- We can access the docs from navbar->help->documentaion
- When there is no storage class, access the storage class creation
  doc from Banner
- Add a question mark icon next to the storage class

Refs: #2149

ChengYanJin committed:
ui/doc: add translations

Refs: #2149

ChengYanJin committed:
ui/doc: add a link to documentaion for volume creation

Refs: #2149

ChengYanJin committed:
ui/doc: fix the bug when the content of tooltip is empty but still display

- the bug was imported by adding the <TooltipContent> style

Refs: #2149

bert-e commented:
# Hello chengyanjin,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/2.5/improvement/2149-documentation-access-from-the-UI` with contents from `improvement/2149-documentation-access-from-the-UI`
and `development/2.5`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/2.5/improvement/2149-documentation-access-from-the-UI origin/development/2.5
 $ git merge origin/improvement/2149-documentation-access-from-the-UI
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.5/improvement/2149-documentation-access-from-the-UI
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort

slaperche-scality commented on ui/src/translations/en.json:
```diff
@@ -131,7 +131,10 @@
+  "import_solution_hint": "You should import a solution to prepare your environment",
+  "documentation": "Documentation",
+  "storage_class_creation_documentation": "storage class creation documentation",
+  "volume_creation_documentation": "volume creation documentation"
```
Given that those icon are located next to the item they document, maybe `Access to the documentation`/`Accéder à la documentation` is enough.

bert-e commented:
# History mismatch



Merge commit #bb1e6074de5eae475d7c965682ffc9a269acd77f on the integration branch
`w/2.5/improvement/2149-documentation-access-from-the-UI` is merging a branch which is neither the current
branch `improvement/2149-documentation-access-from-the-UI` nor the development branch
`development/2.5`.

It is likely due to a rebase of the branch `improvement/2149-documentation-access-from-the-UI` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

ChengYanJin commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/2.5/improvement/2149-documentation-access-from-the-UI` with contents from `improvement/2149-documentation-access-from-the-UI`
and `development/2.5`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/2.5/improvement/2149-documentation-access-from-the-UI origin/development/2.5
 $ git merge origin/improvement/2149-documentation-access-from-the-UI
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.5/improvement/2149-documentation-access-from-the-UI
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort

ChengYanJin committed:
ui/doc: change the cluster status icon from question to info

We want to keep the consistency of the icon.
since we already have quesion icon in the navbar, it should be better we use another icon.

bert-e commented:
# History mismatch



Merge commit #2888830e6c798cd865817e27ce88ec068443e0d9 on the integration branch
`w/2.5/improvement/2149-documentation-access-from-the-UI` is merging a branch which is neither the current
branch `improvement/2149-documentation-access-from-the-UI` nor the development branch
`development/2.5`.

It is likely due to a rebase of the branch `improvement/2149-documentation-access-from-the-UI` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

ChengYanJin commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/2.5/improvement/2149-documentation-access-from-the-UI` with contents from `improvement/2149-documentation-access-from-the-UI`
and `development/2.5`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/2.5/improvement/2149-documentation-access-from-the-UI origin/development/2.5
 $ git merge origin/improvement/2149-documentation-access-from-the-UI
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.5/improvement/2149-documentation-access-from-the-UI
```

slaperche-scality commented on ui/src/containers/ClusterMonitoring.js:
```diff
@@ -222,7 +222,7 @@ const ClusterMonitoring = props => {
             }
           >
-            <QuestionMarkIcon className="fas fa-question-circle" />
+            <QuestionMarkIcon className="fas fa-info-circle" />
```
Not related to this PR (and can't be fixed here, I supposed it's coming from core-ui), but the naming seems poor to me: we call `QuestionMarkIcon` something that's not always a question mark icon…

ChengYanJin replied:
Forgot to change the name `QuestionMarkIcon` to `InformationMarkIcon`, my bad.
Sure, it's not related to this PR, but it's a tiny change in order to keep the consistency for the icons. 
And just FYI, the icon we import directly from the font-awsome library ;)

gdemonet reviewed (commented):
Code LGTM, will test soon :)

gdemonet commented:
/create_pull_requests

bert-e commented:
# History mismatch



Merge commit #7fb97235d2b10710ed79db6b8c8d7652f1c4483d on the integration branch
`w/2.5/improvement/2149-documentation-access-from-the-UI` is merging a branch which is neither the current
branch `improvement/2149-documentation-access-from-the-UI` nor the development branch
`development/2.5`.

It is likely due to a rebase of the branch `improvement/2149-documentation-access-from-the-UI` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **create_pull_requests**

gdemonet commented:
/status

bert-e commented:
# Status





*Status report is not available.*






*The following options are set:* **create_pull_requests**

bert-e commented:
# History mismatch



Merge commit #7fb97235d2b10710ed79db6b8c8d7652f1c4483d on the integration branch
`w/2.5/improvement/2149-documentation-access-from-the-UI` is merging a branch which is neither the current
branch `improvement/2149-documentation-access-from-the-UI` nor the development branch
`development/2.5`.

It is likely due to a rebase of the branch `improvement/2149-documentation-access-from-the-UI` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **create_pull_requests**

ChengYanJin commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.






*The following options are set:* **create_pull_requests**

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/2.5/improvement/2149-documentation-access-from-the-UI` with contents from `improvement/2149-documentation-access-from-the-UI`
and `development/2.5`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/2.5/improvement/2149-documentation-access-from-the-UI origin/development/2.5
 $ git merge origin/improvement/2149-documentation-access-from-the-UI
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.5/improvement/2149-documentation-access-from-the-UI
```





*The following options are set:* **create_pull_requests**

ChengYanJin commented:
/status

bert-e commented:
# Status





*Status report is not available.*






*The following options are set:* **create_pull_requests**

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/2149-documentation-access-from-the-UI` into
`development/2.4`
* [w/2.5/improvement/2149-documentation-access-from-the-UI](https://github.com/scality/metalk8s/compare/development/2.5...w/2.5/improvement/2149-documentation-access-from-the-UI) will be merged into `development/2.5` (pull request #2180)



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`




*Follow* integration pull requests if you would like to be notified of
build statuses by email.







*The following options are set:* **create_pull_requests**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort










*The following options are set:* **create_pull_requests**

ChengYanJin committed:
ui/doc: add url_doc inside the configMap

Refs: #2149

bert-e commented:
# Conflict




A conflict has been raised during the update of
integration branch `w/2.5/improvement/2149-documentation-access-from-the-UI` with contents from `improvement/2149-documentation-access-from-the-UI`
and `development/2.5`.

Please resolve the conflict on **the integration branch** (`w/2.5/improvement/2149-documentation-access-from-the-UI`).


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout w/2.5/improvement/2149-documentation-access-from-the-UI
 $ git pull  # or "git reset --hard origin/w/2.5/improvement/2149-documentation-access-from-the-UI"
 $ git merge origin/development/2.5
 $ # <intense conflict resolution>
 $ git commit
 $ git merge origin/improvement/2149-documentation-access-from-the-UI
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.5/improvement/2149-documentation-access-from-the-UI
```





*The following options are set:* **create_pull_requests**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort










*The following options are set:* **create_pull_requests**

ChengYanJin committed:
ui/doc: add cypress test for the documentation link

Refs: #2149

bert-e commented:
# Conflict




A conflict has been raised during the update of
integration branch `w/2.5/improvement/2149-documentation-access-from-the-UI` with contents from `improvement/2149-documentation-access-from-the-UI`
and `development/2.5`.

Please resolve the conflict on **the integration branch** (`w/2.5/improvement/2149-documentation-access-from-the-UI`).


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout w/2.5/improvement/2149-documentation-access-from-the-UI
 $ git pull  # or "git reset --hard origin/w/2.5/improvement/2149-documentation-access-from-the-UI"
 $ git merge origin/development/2.5
 $ # <intense conflict resolution>
 $ git commit
 $ git merge origin/improvement/2149-documentation-access-from-the-UI
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.5/improvement/2149-documentation-access-from-the-UI
```





*The following options are set:* **create_pull_requests**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort










*The following options are set:* **create_pull_requests**

gdemonet commented on ui/cypress/integration/AccessDocumentation.feature:
```diff
@@ -0,0 +1,4 @@
+Feature: AccessDocumentation
+ Scenario: Access documentation from the Navbar
+     Given I log in
+     Then I access the documentation from the navbar
```
nitpick: `Then`s are more about verifying some fact, not performing an action. Wording could reflect this, by adjusting it slightly to:

```gherkin
    Then I can access the documentation from the Navbar
```

ChengYanJin replied:
It's definitely a good point to improve!!!

gdemonet commented on ui/cypress/integration/common/accessDocumentation.js:
```diff
@@ -0,0 +1,26 @@
+      // win is the the remote page's window object
+      // `cy.stub()` stubs the window.open method
+      cy.stub(win, 'open', url => {
+        // cannot make a HTTP request to the documentation url because cypress doesn't support the `SameSite`
```
I don't really understand this issue, what's this `SameSite` problem?

ChengYanJin replied:
As we discussed, we are not supposed to nest the `cy.resquest` inside the `cy.visit`
I just check if the doc URL is correct. 
For the doc endpoint, we will need a pytest to cover it.

gdemonet commented on ui/cypress/integration/common/accessDocumentation.js:
```diff
@@ -0,0 +1,26 @@
+        //   expect(response.status).to.eq(200);
+        // });
+        // what we can do for the moment is to check the url
+        expect(url).to.have.string('/index.html');
```
Can we check the full path, and that it includes the `/docs` Ingress path as well?

gdemonet reviewed (commented):
Really minor comments, waiting for your answer to give an approval, but looks good to me otherwise :)

bert-e commented:
# History mismatch



Merge commit #18977211ad78e2eea040b92996195260152651df on the integration branch
`w/2.5/improvement/2149-documentation-access-from-the-UI` is merging a branch which is neither the current
branch `improvement/2149-documentation-access-from-the-UI` nor the development branch
`development/2.5`.

It is likely due to a rebase of the branch `improvement/2149-documentation-access-from-the-UI` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **create_pull_requests**

ChengYanJin commented:
/reset

bert-e commented:
# Lossy reset warning



There seems to be *manual* commits on integration branches (e.g. conflict
resolutions) that will be lost if you chose to *reset*.

**You can use the `force_reset` command if you still want me
to delete those branches.**





*The following options are set:* **create_pull_requests**

bert-e commented:
# History mismatch



Merge commit #18977211ad78e2eea040b92996195260152651df on the integration branch
`w/2.5/improvement/2149-documentation-access-from-the-UI` is merging a branch which is neither the current
branch `improvement/2149-documentation-access-from-the-UI` nor the development branch
`development/2.5`.

It is likely due to a rebase of the branch `improvement/2149-documentation-access-from-the-UI` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **create_pull_requests**

ChengYanJin commented:
/status

bert-e commented:
# Status





*Status report is not available.*






*The following options are set:* **create_pull_requests**

bert-e commented:
# History mismatch



Merge commit #18977211ad78e2eea040b92996195260152651df on the integration branch
`w/2.5/improvement/2149-documentation-access-from-the-UI` is merging a branch which is neither the current
branch `improvement/2149-documentation-access-from-the-UI` nor the development branch
`development/2.5`.

It is likely due to a rebase of the branch `improvement/2149-documentation-access-from-the-UI` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **create_pull_requests**

ChengYanJin commented:
/force_reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.






*The following options are set:* **create_pull_requests**

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/2.5/improvement/2149-documentation-access-from-the-UI` with contents from `improvement/2149-documentation-access-from-the-UI`
and `development/2.5`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/2.5/improvement/2149-documentation-access-from-the-UI origin/development/2.5
 $ git merge origin/improvement/2149-documentation-access-from-the-UI
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.5/improvement/2149-documentation-access-from-the-UI
```





*The following options are set:* **create_pull_requests**

ChengYanJin commented:
/status

bert-e commented:
# Status





*Status report is not available.*






*The following options are set:* **create_pull_requests**

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/2149-documentation-access-from-the-UI` into
`development/2.4`
* [w/2.5/improvement/2149-documentation-access-from-the-UI](https://github.com/scality/metalk8s/compare/development/2.5...w/2.5/improvement/2149-documentation-access-from-the-UI) will be merged into `development/2.5` (pull request #2184)



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`




*Follow* integration pull requests if you would like to be notified of
build statuses by email.







*The following options are set:* **create_pull_requests**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort










*The following options are set:* **create_pull_requests**

ChengYanJin commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.






*The following options are set:* **create_pull_requests**

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/2.5/improvement/2149-documentation-access-from-the-UI` with contents from `improvement/2149-documentation-access-from-the-UI`
and `development/2.5`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/2.5/improvement/2149-documentation-access-from-the-UI origin/development/2.5
 $ git merge origin/improvement/2149-documentation-access-from-the-UI
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.5/improvement/2149-documentation-access-from-the-UI
```





*The following options are set:* **create_pull_requests**

ChengYanJin commented:
/status

bert-e commented:
# Status





*Status report is not available.*






*The following options are set:* **create_pull_requests**

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/2149-documentation-access-from-the-UI` into
`development/2.4`
* [w/2.5/improvement/2149-documentation-access-from-the-UI](https://github.com/scality/metalk8s/compare/development/2.5...w/2.5/improvement/2149-documentation-access-from-the-UI) will be merged into `development/2.5` (pull request #2185)



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`




*Follow* integration pull requests if you would like to be notified of
build statuses by email.







*The following options are set:* **create_pull_requests**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort










*The following options are set:* **create_pull_requests**

gdemonet reviewed (dismissed):
LGTM

bert-e commented:
# History mismatch



Merge commit #98b9555192a785020db5d0d1a03f5a3954181740 on the integration branch
`w/2.5/improvement/2149-documentation-access-from-the-UI` is merging a branch which is neither the current
branch `improvement/2149-documentation-access-from-the-UI` nor the development branch
`development/2.5`.

It is likely due to a rebase of the branch `improvement/2149-documentation-access-from-the-UI` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **create_pull_requests**

ChengYanJin commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.






*The following options are set:* **create_pull_requests**

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/2.5/improvement/2149-documentation-access-from-the-UI` with contents from `improvement/2149-documentation-access-from-the-UI`
and `development/2.5`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/2.5/improvement/2149-documentation-access-from-the-UI origin/development/2.5
 $ git merge origin/improvement/2149-documentation-access-from-the-UI
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.5/improvement/2149-documentation-access-from-the-UI
```





*The following options are set:* **create_pull_requests**

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/2149-documentation-access-from-the-UI` into
`development/2.4`
* [w/2.5/improvement/2149-documentation-access-from-the-UI](https://github.com/scality/metalk8s/compare/development/2.5...w/2.5/improvement/2149-documentation-access-from-the-UI) will be merged into `development/2.5` (pull request #2194)



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`




*Follow* integration pull requests if you would like to be notified of
build statuses by email.







*The following options are set:* **create_pull_requests**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort










*The following options are set:* **create_pull_requests**

ChengYanJin commented:
/approve

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.4`

* :heavy_check_mark: `development/2.5`



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **create_pull_requests, approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.4`

* :heavy_check_mark: `development/2.5`



The following branches have **NOT** changed:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`



Please check the status of the associated issue None.

Goodbye chengyanjin.
